.github/workflows/ci.yml

```diff
@@ -34,6 +34,16 @@ jobs:
         with:
           python-version-file: ".python-version"
 
+      - name: Install poetry
+        run: curl -sSL https://install.python-poetry.org | python3 -
+
+      - name: Check pyproject.toml & poetry.lock are in sync
+        run: poetry lock --check
+
+      # We only need to install the dev dependencies, which will install mypy and related packages
+      - name: Install dev dependencies
+        run: poetry install --only=dev
+
```

Comment:
Just an understanding check

We have to install deps in our CI because of the local pre-commit hook you are defining below, which depends on `mypy` and others

Reply:
yep